engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java

```diff
@@ -282,6 +279,25 @@ protected void extractFacetResults() {
 		return SUPPORTED_PROJECTION_CONSTANTS;
 	}
 
+	@Override
+	protected List<IndexManager> getIndexManagers(EntityIndexBinding binding) {
+		List<IndexManager> indexManagers = super.getIndexManagers( binding );
+
+		for ( IndexManager indexManager : indexManagers ) {
+			if ( !( indexManager instanceof DirectoryBasedIndexManager ) ) {
+				//TODO clarify intent:
+				// A) improve the error message so that people understand what they should do
+				// B) Is the point really to not support "directory-based" or rather non-Lucene native based ones?
```

Comment:
But you're right: this comment should disappear eventually (why the hell create a TODO in the code without a corresponding ticket???). I created [HSEARCH-2724](https://hibernate.atlassian.net/browse/HSEARCH-2724).